src/hooks/sounds.tsx

```diff
@@ -35,35 +37,56 @@ export interface ExposedDataWithPlay extends ExposedData {
 
 export interface CustomSoundsState {
   soundConfigs: SoundConfigMap;
+  isLoading: boolean;
 }
 
 export const initialValue: CustomSoundsState = {
   soundConfigs: {},
+  isLoading: false,
 };
 
 export const CustomSoundsContext = createContext<CustomSoundsState>(
   initialValue
 );
 
-export const CustomSoundsProvider: React.FC = ({ children }) => {
+export type CustomSoundsType = {
+  loadingComponent: React.ReactNode;
+  waitTillConfigLoaded: boolean;
+};
+
+export const CustomSoundsProvider: React.FC<CustomSoundsType> = ({
+  loadingComponent,
+  waitTillConfigLoaded,
+  children,
+}) => {
```

Comment:
Setting defaults for the now-optional parameters. Not sure if this will work exactly as is, but something like this:

```suggestion
export const CustomSoundsProvider: React.FC<CustomSoundsProviderProps> = ({
  loadingComponent = 'Loading...',
  waitTillConfigLoaded = false,
  children,
}) => {
```